libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccount.m

```diff
@@ -274,13 +278,18 @@ - (void)setPhoto:(UIImage *)photo {
                     [SFSDKCoreLogger e:[self class] format:@"Unable to remove previous photo from disk: %@", error];
                 }
             }
-            NSData *data = UIImagePNGRepresentation(photo);
-            if (![data writeToFile:photoPath options:NSDataWritingAtomic error:&error]) {
-                [SFSDKCoreLogger e:[self class] format:@"Unable to write photo to disk: %@", error];
+            if (photo) {
+                NSData *data = UIImagePNGRepresentation(photo);
+                if (![data writeToFile:photoPath options:NSDataWritingAtomic error:&error]) {
+                    [SFSDKCoreLogger e:[self class] format:@"Unable to write photo to disk: %@", error];
+                }
             }
             [self willChangeValueForKey:@"photo"];
```

Comment:
We should move the willChange and didChange inside the if then block?

Reply:
Which if then block are you thinking?

Reply:
```
[self willChangeValueForKey:@"photo"];
self->_photo = photo;
[self didChangeValueForKey:@"photo"];
```
This code should be inside the `if (photo) {` block, shouldn't it?

Reply:
@jeanbovet ideally we should be doing this only if the value was set i.e. if there isn't an error.`[data writeToFile:photoPath options:NSDataWritingAtomic error:&error]` Since you are refactoring this piece of code it would be  good to fix this. In the past it's possible that a nil is sent, the value is not set but we still notify the observers.

Reply:
Sure, it makes sense to set it only if there was no error. However, we still need to honor the fact that the user can set a nil photo, right?

Reply:
Updated, let me know what you think